src/Agent.Worker/TaskRunner.cs

```diff
@@ -56,10 +56,11 @@ public async Task RunAsync()
             var handlerFactory = HostContext.GetService<IHandlerFactory>();
 
             // Set the task id and display name variable.
-            ExecutionContext.Variables.Set(Constants.Variables.Task.DisplayName, DisplayName);
-            ExecutionContext.Variables.Set(WellKnownDistributedTaskVariables.TaskInstanceId, Task.Id.ToString("D"));
-            ExecutionContext.Variables.Set(WellKnownDistributedTaskVariables.TaskDisplayName, DisplayName);
-            ExecutionContext.Variables.Set(WellKnownDistributedTaskVariables.TaskInstanceName, Task.Name);
+            var scope =  ExecutionContext.Variables.CreateScope();
```

Comment:
normally I think we would use `using` instead of manually calling dispose. 
```csharp
using(var scope = ExecutionContext.Variables.CreateScope()){
    // things using "scope"
}